test/yz_index_hashtree_eqc.erl

```diff
@@ -0,0 +1,522 @@
+
+    eq(RightDiff, Res).
+
+%% @doc compare - Runs comparison between *local* YZ Idx Hashtree and
+%%      *remote* KV Idx Hashtree.
+-spec compare({ok, tree()}, {ok, tree()}) -> integer().
+compare({ok, YZTreePid}, {ok, KVTreePid}) ->
+    Remote = fun(get_bucket, {L, B}) ->
+                     riak_kv_index_hashtree:exchange_bucket(?TEST_INDEX_N, L, B,
+                                                            KVTreePid);
+                (key_hashes, Segment) ->
+                     riak_kv_index_hashtree:exchange_segment(?TEST_INDEX_N,
+                                                             Segment,
+                                                             KVTreePid);
+                (_, _) ->
+                     ok
+             end,
+
+    AccFun = fun(KeyDiff, Count) ->
+                     lists:foldl(fun(Diff, InnerCount) ->
+                                         case repair(0, Diff) of
+                                             full_repair -> InnerCount + 1;
+                                             _ -> InnerCount
+                                         end
+                                 end, Count, KeyDiff)
+             end,
+
+    yz_index_hashtree:compare(?TEST_INDEX_N, Remote, AccFun, 0, YZTreePid).
+
+%% Property Test
+prop_correct() ->
+    ?FORALL(Cmds,commands(?MODULE, #state{}),
+            aggregate(command_names(Cmds),
+                      ?TRAPEXIT(begin
+                                    {H, S, Res} = run_commands(?MODULE, Cmds),
+
+                                    catch yz_index_hashtree:destroy(
+                                            element(2, S#state.yz_idx_tree)),
+                                    catch riak_kv_index_hashtree:destroy(
+                                            element(2, S#state.kv_idx_tree)),
+
+                                    pretty_commands(?MODULE,
+                                                    Cmds,
+                                                    {H, S, Res},
+                                                    Res == ok)
+                                end))).
+
+%% Private
+-spec set_treedata(obj(), dict()) -> dict().
+set_treedata(RObj, D) ->
+    BKey = eqc_util:get_bkey_from_object(RObj),
+    dict:store(BKey, yz_kv:hash_object(RObj), D).
+
+-spec repair(p(), keydiff()) -> full_repair | tree_repair | failed_repair.
+repair(_, {remote_missing, KeyBin}) ->
+    BKey = binary_to_term(KeyBin),
+    Index = yz_kv:get_index(BKey),
+    case yz_kv:should_index(Index) of
+        true ->
+            full_repair;
+        false ->
+            tree_repair
+    end;
+repair(_Partition, {missing, KeyBin}) ->
+    BKey = binary_to_term(KeyBin),
+    Index = yz_kv:get_index(BKey),
+    case yz_kv:should_index(Index) of
+        true ->
+            full_repair;
+        false ->
+            tree_repair
+    end.
+
+-spec intersection(list(), list()) -> list().
+intersection(List1, List2) ->
```

Comment:
Sorry, more dead code! In the future one should use `sets` or `ordsets` for this anyway.